blockchain/state_migration.go

```diff
@@ -103,12 +103,12 @@ func (bc *BlockChain) migrateState(rootHash common.Hash) (returnErr error) {
 
 	start := time.Now()
 
-	srcCachedDB := bc.StateCache().TrieDB()
-	targetDB := statedb.NewDatabase(&stateTrieMigrationDB{bc.db})
+	srcState := bc.StateCache()
+	dstState := state.NewDatabase(&stateTrieMigrationDB{bc.db})
```

Comment:
I am confused after the rebase. The purpose of this PR is to use fastcache for `dstState`, right? But, this line means you are not using the fastcache.
Even though `dstState` is used in `CheckStateConsistency`, it is db with no cache.

Reply:
This PR makes `CheckStateConsistency` can use a state with a cache for old DB.
For new DB, we should not set the same cache. :)

So this code is right.

Reply:
Do you mean the code below (this is from `CheckStateConsistency`) makes a new fastcache?
```
oldState, err := New(root, oldDB)
newState, err := New(root, newDB)
```
If not, can you tell me which part of the code is using a fastcache?

Reply:
Now I understand :) 
The fastcache is used for `srcState` (in code`srcState := bc.StateCache()`).
Nodes are fetched from either fastcache or `srcState` and stored in `dstState`.